coding-the-legislation/25_vectorial_computing.md

```diff
@@ -20,6 +20,33 @@ array([45, 42, 17])
 
 Basic operations such as `+` or `*` behave the same way on vectors than on numbers, you can thus use them in OpenFisca. However, some operations and structures must be adapted.
 
+## Scalars
+
+As a first-time developer, it is natural to write such a formula
+```py
+def formula(person):
+    rebate = 610
+    return rebate
+```
+Unfortunately, this will generate an error similar to
+> The formula ‘rates_rebate@2016’ should return a Numpy array; instead it returned ‘610.0’ of ’<type ‘float’>’.
+
+Indeed, formulas receive an array of entity (*n* _people_ or *m* _households_ in `country_template`) and OpenFisca expects formulas to return an array of the same shape.
```

Comment:
Link to the `entity` documentation page.

Reply:
Technical precision: what formulas receive is not exactly an array of entity. It's a more complex object that represents a list of entities:  I cannot `map`, `filter` it or even iterate on it.

I don't know how technical we want to be here, but claiming it's an array may be misleading.